dodiscover/replearning/gin.py

```diff
@@ -0,0 +1,64 @@
+"""
+Wrapper for the GIN algorithm in causal-learn
+TODO: Need type hints
+"""
+from causallearn.search.HiddenCausal.GIN.GIN import GIN as GIN_
+
+from pywhy_graphs import CPDAG
+
+class GIN:
+    """
+    Dodiscover wrapper for GIN in causal-learn package
+    """
+    def __init__(self, indep_test_method="kci", alpha=0.05):
+        """
+        Using default parameters from GIN.
+        TODO: Add full set of GIN parameters with default
+        TODO: Add a base class
+        """
+        self.graph_ = None # Should be in a base class
+
+        # GIN default parameters.
+        self.indep_test_method = indep_test_method
+        self.alpha = alpha
+        # The follow objects are specific to causal-learn, perhaps they should
+        # go in a base class too.
+        self.causal_learn_graph = None
+        self.causal_ordering = None
+
+    def _causal_learn_to_pdag(self, cl_graph):
```

Comment:
A help function to turn the output graph of causal learn to a mixed graph.  Kind of a hack.  Might be good to pull this out of the class.

@adam2392 is CPDAG the right output type here?

Reply:
What does the GIN algorithm exactly output? Is it the graph, or an equivalence class?

Reply:
A mixed graph, directed and undirected.   I'd call it a PDAG, though their might be some theoretical issues there.

Reply:
In pywhy-graphs, there is a function called `clearn_arr_to_graph(arr: np.ndarray, arr_idx: List[Node], graph_type: str)`, do you want to try it with:

`clearn_arr_to_graph(clearn_graph_arr_from_causal_learn, arr_idx=names_of_nodes, graph_type='cpdag')` and see if it returns you the right answer?

Reply:
@adam2392 [6380da3](https://github.com/py-why/dodiscover/pull/72/commits/6380da386142732c6c3300dbec655fbd731c37e8)